zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -1129,3 +1142,67 @@ def get_postgresql_databases(cluster_id, alias, environment, region, infrastruct
                 entities.append(replicatype_entity)
 
     return entities
+
+
+@trace(tags={'kubernetes': 'hpa'}, pass_span=True)
+def get_cluster_hpas(kube_client, cluster_id, alias, environment, region, infrastructure_account,
+                     namespace=None, **kwargs) -> list:
+    current_span = extract_span_from_kwargs(**kwargs)
+    entities = []
+
+    hpas = get_all(kube_client, kube_client.get_hpas, namespace, span=current_span)
+
+    for h in hpas:
+        obj = h.obj
+
+        entity = {
+            'id': 'hpa-{}-{}[{}]'.format(h.name, h.namespace, cluster_id),
+            'type': HPA_TYPE,
+            'kube_cluster': cluster_id,
+            'alias': alias,
+            'environment': environment,
+            'created_by': AGENT_TYPE,
+            'infrastructure_account': infrastructure_account,
+            'region': region,
+
+            'hpa_name': h.name,
+            'hpa_namespace': obj['metadata']['namespace'],
+
+            'hpa_desired_replicas': obj['status']['desiredReplicas'],
+            'hpa_current_replicas': obj['status']['currentReplicas']
```

Comment:
Please add `min_replicas`, `max_replicas` and `target_cpu_utilization_percentage` (from `spec`).

Reply:
Not all HPAs will have target CPU utilization. I can parse them if present and add them but I feel the entity should be as consistent as possible.

Reply:
I don't see a technical issue, just that it's kind of misleading because there are cases when the cpu utilization will actually be 0.

Reply:
I see what you mean. I don't know what the best way to handle it. Like we could add a field `cpu_metrics: True|False` but maybe it's simpler to just leave it out of the entity if it's not enabled. It's not gonna be something you filter on in ZMON.

Reply:
I don't see any harm in adding it to the entity only if it's present in the HPA, but you could also leave it out I guess. Replicas I would definitely add because it'd be hard to design meaningful alerts without that information.